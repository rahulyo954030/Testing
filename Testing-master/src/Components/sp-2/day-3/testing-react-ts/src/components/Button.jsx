
import "./button.css"

export const Button = ({variant, colorScheme, children, onClick }) => {
  return (
    <div>
        <button
        data-testid="cButton"
        onClick={onClick}
         className={`btn ${colorScheme || "red"} ${variant || "ghost"}`} 
         >
            {children}
            </button>
    </div>
  )
}
