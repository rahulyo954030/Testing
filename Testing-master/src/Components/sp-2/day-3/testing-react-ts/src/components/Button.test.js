import { render, screen, fireEvent } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer"

import { Button } from "./Button";

describe("Button Tests", function () {
    it("should have button in DOM", function () {
        render(<Button>Click me</Button>); //virtully
        let button = screen.getByText("Click me");
        expect(button).toBeInTheDocument();

        // screen.debug() this use to debug code
    })

    it("should render based on testid", function (){
        render(<Button>Click me</Button>);

        let button= screen.getByTestId("cButton");
        expect(button).toBeInTheDocument();
    });

    it("should render empty button", function () {
        render(<Button></Button>);
        
        let button = screen.getByTestId("cButton");
        expect(button).toBeEmptyDOMElement();
    })
    // xtest and xit will sttop that test case / skip

    //   it.only will only that test case 

    it("should change theme" , function (){
            render(<App></App>)

            let h3= screen.getByText("Theme is light")
            expect(h3).toHaveTextContent("light")


            let themeButton = screen.getByText("change theme")
            fireEvent.click(themeButton)

            expect(h3).toHaveTextContent("dark")
        })

        it("should call given function", function (){
            const mockfn = jest.fn();
            // jest known when called it how many time called it
            render(<Button onClick={mockfn}>Click me</Button>)
                // function call is or not
            const btn = screen.getByText("Click me");
            // const btn1 = screen.getByTestId("cButton");
            fireEvent.click(btn)
            fireEvent.click(btn);

            expect(mockfn).toBeCalledTimes(2)
        })

        // snapshot
        it.only("should match snpshopt", function (){
            const tree = renderer
            .create(
                <Button colorScheme={"green"} variant="ghost">
                    Add Counter
                </Button>
            )
            .toTree();
            
            expect(tree).toMatchSnapshot();
        })

        it("should show correct red classes", function (){
            render(<Button colorScheme={"red"}>Click me</Button>);

            const buttn = screen.getByTestId("cButton");
            expect(buttn).toHaveClass("red")
        })

        it("should show correct green classes", function (){
            render(<Button colorScheme={"green"}>Click me</Button>);

            const buttn = screen.getByTestId("cButton");
            expect(buttn).toHaveClass("green")
        })
    
})

// beforeAll(() =>{
//     // connect to a db
// })
// beforeEach(() =>{
//     // visit("deployed link")
// })
// afterAll(() =>{
//     // disconnect from db
// })
// afterEach(() =>{
//     // 
// })

