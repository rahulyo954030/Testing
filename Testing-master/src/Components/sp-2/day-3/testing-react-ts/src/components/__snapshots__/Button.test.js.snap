// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests should match snpshopt 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": "Add Counter",
    "colorScheme": "green",
    "variant": "ghost",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="btn green ghost"
        data-testid="cButton"
      >
        Add Counter
      </button>,
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "Add Counter",
          "className": "btn green ghost",
          "data-testid": "cButton",
          "onClick": undefined,
        },
        "rendered": Array [
          "Add Counter",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
